deptrac.config.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+use Internal\Qossmic\Deptrac\IgnoreDependenciesOnContract;
+use Internal\Qossmic\Deptrac\IgnoreDependenciesOnShouldNotHappenException;
+use Qossmic\Deptrac\Contract\Analyser\ProcessEvent;
+use Qossmic\Deptrac\Core\Layer\Collector\CollectorType;
+use Qossmic\Deptrac\Supportive\Config\DeptracConfig;
+use Qossmic\Deptrac\Supportive\DependencyInjection\EmitterType;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+return static function (DeptracConfig $config, ContainerConfigurator  $containerConfigurator): void {
+    $services = $containerConfigurator->services();
+    $services->set(IgnoreDependenciesOnContract::class)->tag('kernel.event_listener', ['event' => ProcessEvent::class]);
+    $services->set(IgnoreDependenciesOnShouldNotHappenException::class)->tag('kernel.event_listener', ['event' => ProcessEvent::class]);
+
+    $config->paths(['./src']);
+    $config->baseline('./deptrac.baseline.yaml');
+    // analyser
+    $config->analyser(
+        EmitterType::CLASS_TOKEN,
+        EmitterType::CLASS_SUPERGLOBAL_TOKEN,
+        EmitterType::FILE_TOKEN,
+        EmitterType::FUNCTION_TOKEN,
+        EmitterType::FUNCTION_SUPERGLOBAL_TOKEN,
+        EmitterType::FUNCTION_CALL,
+    );
+
+    // layer
+    $analyser = $config->layer('Analyser');
+    $analyser->collector(CollectorType::TYPE_DIRECTORY)->value('src/Core/Analyser/.*');
+
+    $ast = $config->layer('Ast');
+    $ast->collector(CollectorType::TYPE_DIRECTORY)->value('src/Core/Ast/.*');
+    $ast->collector(CollectorType::TYPE_CLASS_NAME)->value('^PHPStan\\\\PhpDocParser\\\\.*')->private();
```

Comment:
WHy do we need `\\\\` here and not just `\\`?

Reply:
Just check this out: https://3v4l.org/VmMtn.

Reply:
I understand, The string I look for is `PHPStan\PhpDocParser\`, therefore I should be good with 2 backslashes, not 4, hence my question.

Reply:
The regex we want to have is '^PHPStan\\PhpDocParser\\.*', right? 
But php escapes `\\` to `\` inside of the stored string. Therefore we need to escape the escape symbol to get the regex we want.

https://3v4l.org/u61CB

Is this understandable?

Reply:
https://www.php.net/manual/en/regexp.reference.escape.php also explains it.

Reply:
Oh, now I see. Frick, that sucks. But there is nothing to be done about it... We just need to make sure to provide sufficient guidance and documentation.

Reply:
We could offer a function to help ease the pain of using regex for configurations: https://3v4l.org/W5VjK. This way we could also get rid of the `ClassNameRegexCollector`, because the `ClassCollector` would also use the `Regex`-Class. It's inspired by the `services()` HelperFunction symfony offers to configure the container.